controller/konnect/konnectextension_controller.go

```diff
@@ -201,24 +271,44 @@ func (r *KonnectExtensionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 	var apiAuth konnectv1alpha1.KonnectAPIAuthConfiguration
 	err = r.Client.Get(ctx, apiAuthRef, &apiAuth)
-	if requeue, res, retErr := handleAPIAuthStatusCondition(ctx, r.Client, &ext, apiAuth, err); requeue {
+	if requeue, res, retErr := handleAPIAuthStatusCondition(
+		ctx,
+		r.Client,
+		&ext,
+		apiAuth,
+		err,
+		readyCondition,
+	); requeue {
 		return res, retErr
 	}
 
+	apiAuthConfigValidCond := metav1.Condition{
+		Type:    konnectv1alpha1.KonnectEntityAPIAuthConfigurationValidConditionType,
+		Status:  metav1.ConditionTrue,
+		Reason:  konnectv1alpha1.KonnectEntityAPIAuthConfigurationReasonValid,
+		Message: "APIAuthConfiguration is valid",
+	}
+
 	token, err := getTokenFromKonnectAPIAuthConfiguration(ctx, r.Client, &apiAuth)
 	if err != nil {
-		if res, errStatus := patch.StatusWithCondition(
-			ctx, r.Client, &ext,
-			konnectv1alpha1.KonnectEntityAPIAuthConfigurationValidConditionType,
-			metav1.ConditionFalse,
-			konnectv1alpha1.KonnectEntityAPIAuthConfigurationReasonInvalid,
-			err.Error(),
-		); errStatus != nil || !res.IsZero() {
+		apiAuthConfigValidCond.Status = metav1.ConditionFalse
+		apiAuthConfigValidCond.Reason = konnectv1alpha1.KonnectEntityAPIAuthConfigurationReasonInvalid
+		apiAuthConfigValidCond.Message = err.Error()
+		if res, updated, errStatus := patch.StatusWithConditions(
+			ctx,
+			r.Client,
+			&ext,
+			readyCondition,
+			apiAuthConfigValidCond,
+		); errStatus != nil || updated || !res.IsZero() {
 			return res, errStatus
 		}
+		log.Debug(logger, "token retrieval failed")
 		return ctrl.Result{}, err
 	}
 
+	log.Debug(logger, "token retrieved")
```

Comment:
Suggestion might not compile but I'd suggest adding this information to the log entry.

```suggestion
	log.Debug(logger, "API token retrieved from KonnectAPIAuthConfiguration", "apiAuthRef", apiAuth.Name)
```